src/java/org/apache/cassandra/tcm/CMSOperations.java

```diff
@@ -243,4 +244,16 @@ public void unregisterLeftNodes(List<String> nodeIdStrings)
             cms.commit(new Unregister(nodeId, EnumSet.of(NodeState.LEFT)));
         }
     }
+
+    @Override
+    public void accordMarkStale(List<String> nodeIdStrings)
+    {
+        List<NodeId> nodeIds = nodeIdStrings.stream().map(NodeId::fromString).collect(Collectors.toList());
+
+        for (NodeId nodeId : nodeIds)
```

Comment:
This is going to bump the epoch for each node in the list. I left a more detailed comment on `AccordMarkStale` but tl;dr I would make the transformation mark them all stale in one hit.